### PR TITLE
Arsamme/create musics list fragment

### PR DESCRIPTION
**Pull request by aliraghebi**
added RecyclerView in Musics List Fragment to show musics
and also showing fragment in activity into framelayout(temp)

### CODE REVIEW

**Comment by aliraghebi** [comment]
@karammi thanks, when using databinding in kotlin project android studio suggests adding that plugin.


Kapt can process kotlin files while annoationProcessor can not. So if you are using kotlin you have to use kapt. If none of the files you are processing with butterkife and dagger are written in kotlin you still use annotationProcessor. If you have a mix of kotlin and Java files kapt can handle both.

File: app/build.gradle [hunk 1]
```diff
@@ -27,6 +28,10 @@ android {
         sourceCompatibility = '1.8'
         targetCompatibility = '1.8'
     }
+    viewBinding {
+        enabled = true
+    }
+    dataBinding.enabled = true
```
Comment (helloseyedjafari) [inline app/build.gradle]: lotfan databinding ezafe nakonim be proje

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicItemViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package com.worldsnas.starplayer.view.musics_list
+
+import androidx.databinding.ObservableField
+import com.worldsnas.starplayer.model.Music
+
+
+class MusicItemViewModel(private val music: Music,private val listener:Listener) {
+    var musicTitle: ObservableField<String> = ObservableField(music.title)
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicItemViewModel.kt]: be jaye ina mitonim az LiveData estefade konim!

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.worldsnas.starplayer.core.BaseViewHolder
+import com.worldsnas.starplayer.databinding.ItemMusicBinding
+import com.worldsnas.starplayer.model.Music
+
+class MusicsListAdapter(private val context: Context, private val musics: ArrayList<Music>) :
+        RecyclerView.Adapter<BaseViewHolder>() {
+
+    inner class MusicListItemViewHolder(private val mBinding: ItemMusicBinding) : BaseViewHolder(mBinding.root), MusicItemViewModel.Listener {
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListAdapter.kt]: niazi nist `inner` bashe

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        private var mMusicViewModel: MusicItemViewModel? = null
+        override fun onBind(position: Int) {
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListAdapter.kt]: bejaye `position` mitonid khode object ro pass bedid `music` intori dige niazi be ham be inner bodan nist
toye viewHolder ham position ro darim niaz nist az biron begirim

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListFragment.kt [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+import com.worldsnas.starplayer.R
+import com.worldsnas.starplayer.databinding.FragmentMusicsListBinding
+import com.worldsnas.starplayer.model.Music
+
+class MusicsListFragment : Fragment() {
+
+    private var _binding: FragmentMusicsListBinding? = null
+    private val binding get() = _binding!!
+
+    override fun onCreate(savedInstanceState: Bundle?) {
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListFragment.kt]: fek mikonam niazi behesh nist

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListFragment.kt [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+import android.view.ViewGroup
+
+import com.worldsnas.starplayer.R
+import com.worldsnas.starplayer.databinding.FragmentMusicsListBinding
+import com.worldsnas.starplayer.model.Music
+
+class MusicsListFragment : Fragment() {
+
+    private var _binding: FragmentMusicsListBinding? = null
+    private val binding get() = _binding!!
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListFragment.kt:16]: in kare jalebie 
dost dashtam!
afarin

File: app/src/main/res/layout/activity_main.xml [hunk 7]
```diff
@@ -1,9 +1,16 @@
 <?xml version="1.0" encoding="utf-8"?>
+<layout>
 <androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (helloseyedjafari) [inline app/src/main/res/layout/activity_main.xml]: che niazi be in `ConstraintLayout` bironi hast?
benazaram estefade i ham nemishe

File: dependencies.gradle [hunk 8]
```diff
@@ -18,7 +17,7 @@ ext {
             googleMaterialVersion = "1.0.0-rc01",
             pagingComponentVersion = "1.0.0-alpha6",
             constraintLayoutVersion = "1.1.3",
-            archComponentVersion = "2.0.0-rc01",
+            archComponentVersion = "2.2.0",
```
Comment (helloseyedjafari) [inline dependencies.gradle:20]: mamnonam az update

**Review by helloseyedjafari — CHANGES_REQUESTED** [review]
kheili khobe

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListFragment.kt (hunk 5 above)
Comment (aliraghebi) [reply app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListFragment.kt]: Be chi? Oon dota variable binding?

File: app/src/main/res/layout/activity_main.xml (hunk 7 above)
Comment (aliraghebi) [reply app/src/main/res/layout/activity_main.xml]: Roo main activity kar nakardam faghat khastam fragment ro neshon bedam natije ro bebinam

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListFragment.kt (hunk 5 above)
Comment (helloseyedjafari) [reply app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListFragment.kt]: roye khat comment gozashtam
yani khate 18

File: app/src/main/res/layout/activity_main.xml (hunk 7 above)
Comment (helloseyedjafari) [reply app/src/main/res/layout/activity_main.xml]: hich iradi nadare
bi zahmat eslahesh kon

**Comment by aliraghebi** [comment]
@worldsnas would you please check changes and review?

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListFragment.kt [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+import android.view.View
+import android.view.ViewGroup
+
+import com.worldsnas.starplayer.R
+import com.worldsnas.starplayer.databinding.FragmentMusicsListBinding
+import com.worldsnas.starplayer.model.Music
+
+class MusicsListFragment : Fragment() {
+
+    private var _binding: FragmentMusicsListBinding? = null
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListFragment.kt:15]: in bayad null beshe to `onDestroyView()`

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListAdapter.kt [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.worldsnas.starplayer.core.BaseViewHolder
+import com.worldsnas.starplayer.databinding.ItemMusicBinding
+import com.worldsnas.starplayer.model.Music
+
+class MusicsListAdapter(private val context: Context, private val musics: ArrayList<Music>) :
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListAdapter.kt]: lotfan be adapter `context` pass nadid
kasi context bekhad mitone az `ViewHolder` begire

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListAdapter.kt [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+    RecyclerView.Adapter<BaseViewHolder<Music>>() {
+
+    class MusicListItemViewHolder(private val mBinding: ItemMusicBinding) :
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicsListAdapter.kt]: man basham baraye rahat khondan in class ro biron e adapter misazam 
masalan payine hamin file

File: app/src/main/res/layout/activity_main.xml [hunk 12]
```diff
@@ -6,4 +6,9 @@
     android:layout_height="match_parent"
     tools:context=".MainActivity">
```
Comment (helloseyedjafari) [inline app/src/main/res/layout/activity_main.xml]: chera bayad to in file `constraintlayout` dashte bashim?

File: app/src/main/res/layout/fragment_musics_list.xml [hunk 13]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (helloseyedjafari) [inline app/src/main/res/layout/fragment_musics_list.xml]: ye `recyclerView` tanha ham karemono rah miandaze na?

File: app/src/main/res/layout/item_music.xml [hunk 14]
```diff
@@ -25,17 +25,19 @@
...
     <TextView
-        android:id="@+id/txt_music_singer"
+        android:id="@+id/txt_artist"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
         android:layout_marginStart="8dp"
         android:layout_marginTop="4dp"
         android:textColor="@color/textDarkGrey"
         android:textSize="13sp"
+        android:text="@{viewModel.musicArtist}"
```
Comment (helloseyedjafari) [inline app/src/main/res/layout/item_music.xml]: dige `DataBinding` nadarim
in fek konam kar nemikone na?

**Review by helloseyedjafari — CHANGES_REQUESTED** [review]
good progress

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicItemViewModel.kt (hunk 2 above)
Comment (helloseyedjafari) [reply app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicItemViewModel.kt]: in chi shod?

**Review by helloseyedjafari — APPROVED** [review]
well done

**Pull request merged**